runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -287,5 +319,76 @@ MM_IndexableObjectAllocationModel::layoutDiscontiguousArraylet(MM_EnvironmentBas
 	return spine;
 }
 
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
+#if !defined(LINUX) || !defined(J9VM_ENV_DATA64) || defined(OMRZTPF)
+/* Double map is only supported on LINUX 64 bit Systems for now */
+#error "Platform not supported by Double Map API"
+#endif /* !definedLINUX || !defined(J9VM_ENV_DATA64) || defined(OMRZTPF) */
+void* 
+MM_IndexableObjectAllocationModel::doubleMapArraylets(MM_EnvironmentBase* env, J9Object *objectPtr) 
+{
+	MM_GCExtensions *extensions = MM_GCExtensions::getExtensions(env);
+	J9JavaVM *javaVM = extensions->getJavaVM();
+	PORT_ACCESS_FROM_ENVIRONMENT(env);
+
+	GC_ArrayletLeafIterator arrayletLeafIterator(javaVM, (J9IndexableObject*)objectPtr);
+	UDATA arrayletLeafCount = arrayletLeafIterator.getNumLeafs();
+	MM_Heap *heap = extensions->getHeap();
+
+	void *result = NULL;
+
+	void *leaves[ARRAYLET_ALLOC_THRESHOLD];
```

Comment:
you can bring the definition of ARRAYLET_ALLOC_THRESHOLD here (a line above), even if its is #define
or have it const int.